cpp/src/arrow/engine/substrait/options.cc

```diff
@@ -166,6 +171,57 @@ class DefaultExtensionProvider : public BaseExtensionProvider {
                                      named_tap_rel.name(), std::move(renamed_schema)));
     return RelationInfo{{std::move(decl), std::move(renamed_schema)}, std::nullopt};
   }
+
+  Result<RelationInfo> MakeSegmentedAggregateRel(
```

Comment:
I don't think this belongs in options.cc - this is not related to "options" really

Reply:
There are other `Make*Rel` functions already in `options.cc`, so I think such refactoring could be deferred to a separate PR.